syntax = "proto3";

package mes_grpc;

option go_package = "proto/;mes_grpc";

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

service MessengerServer {
  rpc SendMessage (Data) returns (Ack) {}

  rpc ReadMessages (google.protobuf.Empty) returns (stream Message) {}
}

message Data {
  string author = 1;
  string text = 2;
}

message Ack {
  google.protobuf.Timestamp sendTime = 1;
}

message Message {
  string author = 1;
  string text = 2;
  google.protobuf.Timestamp sendTime = 3;
}
